File: mini_bot_trading.py
import ccxt
import time

# Ganti dengan API key dan secret kamu sendiri!
API_KEY = 'ISI_API_KEY_KAMU'
API_SECRET = 'ISI_API_SECRET_KAMU'

def main():
    exchange = ccxt.binance({
        'apiKey': API_KEY,
        'secret': API_SECRET,
        'enableRateLimit': True,
    })

    symbol = 'BTC/USDT'  # Pasangan trading
    amount = 0.001       # Jumlah BTC yang mau dibeli / dijual

    # Harga beli dan jual limit (contoh)
    buy_price = 25000    # USDT
    sell_price = 26000   # USDT

    print(f"Pasang order LIMIT BUY {amount} {symbol} di harga {buy_price} USDT...")
    buy_order = exchange.create_limit_buy_order(symbol, amount, buy_price)
    print(f"Order BUY ID: {buy_order['id']}")

    print(f"Pasang order LIMIT SELL {amount} {symbol} di harga {sell_price} USDT...")
    sell_order = exchange.create_limit_sell_order(symbol, amount, sell_price)
    print(f"Order SELL ID: {sell_order['id']}")

    # Tunggu 10 detik, lalu cek status order
    time.sleep(10)
    print("\nCek status order...")

    buy_status = exchange.fetch_order(buy_order['id'], symbol)
    sell_status = exchange.fetch_order(sell_order['id'], symbol)

    print(f"Status BUY order: {buy_status['status']}")
    print(f"Status SELL order: {sell_status['status']}")

    # Jika order belum terpenuhi, batalkan
    if buy_status['status'] != 'closed':
        print("Batalin order BUY...")
        exchange.cancel_order(buy_order['id'], symbol)

    if sell_status['status'] != 'closed':
        print("Batalin order SELL...")
        exchange.cancel_order(sell_order['id'], symbol)

    print("Selesai!")

if __name__ == '__main__':
    main()

File: mini_bot_trading.js
const ccxt = require('ccxt');

const API_KEY = 'ISI_API_KEY_KAMU';
const API_SECRET = 'ISI_API_SECRET_KAMU';

async function main() {
    const exchange = new ccxt.binance({
        apiKey: API_KEY,
        secret: API_SECRET,
        enableRateLimit: true,
    });

    const symbol = 'BTC/USDT';
    const amount = 0.001; // Jumlah BTC
    const buyPrice = 25000;
    const sellPrice = 26000;

    try {
        console.log(`Pasang LIMIT BUY ${amount} BTC di ${buyPrice} USDT...`);
        const buyOrder = await exchange.createLimitBuyOrder(symbol, amount, buyPrice);
        console.log(`✅ Order BUY ID: ${buyOrder.id}`);

        console.log(`Pasang LIMIT SELL ${amount} BTC di ${sellPrice} USDT...`);
        const sellOrder = await exchange.createLimitSellOrder(symbol, amount, sellPrice);
        console.log(`✅ Order SELL ID: ${sellOrder.id}`);

        // Tunggu 10 detik
        await new Promise(resolve => setTimeout(resolve, 10000));

        // Cek status order
        console.log(`\n📡 Cek status order...`);
        const buyStatus = await exchange.fetchOrder(buyOrder.id, symbol);
        const sellStatus = await exchange.fetchOrder(sellOrder.id, symbol);

        console.log(`Status BUY: ${buyStatus.status}`);
        console.log(`Status SELL: ${sellStatus.status}`);

        // Batalin jika belum tereksekusi
        if (buyStatus.status !== 'closed') {
            console.log('❌ Cancel BUY order...');
            await exchange.cancelOrder(buyOrder.id, symbol);
        }

        if (sellStatus.status !== 'closed') {
            console.log('❌ Cancel SELL order...');
            await exchange.cancelOrder(sellOrder.id, symbol);
        }

        console.log('✅ Selesai!');

    } catch (err) {
        console.error('⚠️ Error:', err.message || err);
    }
}

main();
